Judge never outputs a status in some cases

https://dodona.ugent.be/en/submissions/3994014/

```
Judge exited with status code 1.
Standard Error:
status enum requires also human
Standard Output:
{
 "command": "start-judgement"
}{
 "command": "start-tab",
 "hidden": true,
 "title": "Compiler"
}{
 "command": "append-message",
 "message": {
  "description": "Je code kon niet worden gecompileerd en bijgevolg niet worden getest. De compiler rapporteerde .",
  "format": "callout"
 }
}{
 "command": "start-context"
}{
 "command": "start-testcase",
 "description": {
  "description": "Compilatiefout: De naam van je klasse hoort \"Basket\" te zijn.",
  "format": "plain"
 }
}{
 "command": "append-message",
 "message": {
  "description": "Basket.java:5: error: class BasketAnalysis is public, should be declared in a file named BasketAnalysis.java\npublic class BasketAnalysis {\n       ^",
  "format": "code"
 }
}{
 "command": "annotate-code",
 "row": 4,
 "text": "De naam van je klasse hoort \"Basket\" te zijn.",
 "column": 7,
 "type": "error"
}{
 "command": "close-testcase",
 "accepted": false
}{
 "command": "close-context"
}{
 "command": "start-context"
}{
 "command": "start-testcase",
 "description": {
  "description": "Compilatiefout: De naam van je klasse hoort \"Basket\" te zijn.",
  "format": "plain"
 }
}{
 "command": "append-message",
 "message": {
  "description": "Basket.java:5: error: class BasketAnalysis is public, should be declared in a file named BasketAnalysis.java\npublic class BasketAnalysis {\n       ^",
  "format": "code"
 }
}{
 "command": "annotate-code",
 "row": 4,
 "text": "De naam van je klasse hoort \"Basket\" te zijn.",
 "column": 7,
 "type": "error"
}{
 "command": "close-testcase",
 "accepted": false
}{
 "command": "close-context"
}{
 "command": "start-context"
}{
 "command": "start-testcase",
 "description": {
  "description": "Compilatiefout",
  "format": "plain"
 }
}{
 "command": "append-message",
 "message": {
  "description": "Basket.java:6: error: cannot access Basket\n    private ArrayList<Basket> baskets;\n                      ^\nBasket.java\n    file does not contain class Basket\n    Please remove or make sure it appears in the correct subdirectory of the sourcepath.",
  "format": "code"
 }
}{
 "command": "annotate-code",
 "row": 5,
 "text": "cannot access Basket",
 "column": 22,
 "type": "error"
}{
 "command": "close-testcase",
 "accepted": false
}{
 "command": "close-context"
}{
 "command": "close-tab"
}
```

```java
public class A {
    private static B b;
}
```

in bestand `B.java` geeft:

```
[16:47:52] ~ $ javac B.java
B.java:1: error: class A is public, should be declared in a file named A.java
public class A {
       ^
       ./B.java:1: error: class A is public, should be declared in a file named A.java
public class A {
       ^
B.java:3: error: cannot access B
        private static B b;
                       ^
  bad source file: ./B.java
    file does not contain class B
    Please remove or make sure it appears in the correct subdirectory of the sourcepath.
```

--- a/run
+++ b/run
@@ -155,8 +155,9 @@ parse_compilation_errors() {
 compilation_failed() {
     compilation="$1"
     callout="$2"
-    human="$3"
-    target="$4"
+    ifcounts="$3"
+    human="$4"
+    target="$5"
 
     # Counting compilation errors and warning.
     compile_error_count="$(compilation_error_count < "$compilation")"
@@ -181,7 +182,9 @@ compilation_failed() {
         && described_both_count="$described_error_count en $described_warning_count" \
         || described_both_count="$described_error_count$described_warning_count"
 
-    dodona append-message -f callout -d "$(printf "$callout" "$described_both_count")"
+    [ -n "$described_both_count" ] \
+        && callout="$(printf "%s $ifcounts" "$callout" "$described_both_count")"
+    dodona append-message -f callout -d "$callout"
 
     if grep -q '^package' "$filename"; then
         dodona append-message -f callout -d 'Ben je zeker dat je de ingediende klasse in het default package plaatste?'
@@ -191,7 +194,7 @@ compilation_failed() {
     sed 's_.*/\([^/]*.java\)_\1_' "$compilation" | parse_compilation_errors "$target"
 
     dodona close-tab # -b "$compile_errwarn_sum" TODO https://github.ugent.be/dodona/dodona/pull/1051
-    dodona close-judgement -A -e 'compilation error' -h "$(printf "$human" "$described_error_count")"
+    dodona close-judgement -A -e 'compilation error' -h "$(printf "$human" "${described_error_count:-Compilatiefout}")"
     exit 0
 }
 
@@ -216,6 +219,7 @@ compilation_error() {
 if ! find . -name '*.java' | xargs --no-run-if-empty javac -cp ".:${worklibs}:${testlibs}" -d . -sourcepath . > "$compilation" 2>&1; then
     compilation_failed "$compilation" \
         'Er ging iets mis tijdens het compileren van de startcode voor deze oefening. Contacteer je lesgever.' \
+        '' \
         'Fout in de startcode' \
         'staff'
 fi
@@ -228,7 +232,8 @@ compile_opts="-Xlint:all"
 [ "$allow_compilation_warnings" == 'true' ] || compile_opts+=' -Werror'
 if ! javac -cp ".:${worklibs}" ${compile_opts} "$filename" > "$compilation" 2>&1; then
     compilation_failed "$compilation" \
-        "Je code kon niet worden gecompileerd en bijgevolg niet worden getest. De compiler rapporteerde %s." \
+        'Je code kon niet worden gecompileerd en bijgevolg niet worden getest.' \
+        'De compiler rapporteerde %s.' \
         "%s" \
         'student'
 fi
@@ -245,7 +250,8 @@ fi
 # Compiling the tests
 if ! find "$resources" -name '*.java' | xargs javac -cp ".:${resources}:${worklibs}:${testlibs}:judge.jar" -d . -sourcepath "$resources" > "$compilation" 2>&1; then
     compilation_failed "$compilation" \
-        'Er ging iets mis tijdens het compileren van de testen voor deze oefening. Contacteer je lesgever.' \
+        'Er ging iets mis tijdens het compileren van de testen voor deze oefening.' \
+        '' \
         'Fout in de testen' \
         'staff'
 fi

